src/julia-parser.scm

```diff
@@ -2073,7 +2073,7 @@
   (string.join (string-split s a) b))
 
 (define (ends-interpolated-atom? c)
-  (or (opchar? c) (never-identifier-char? c)))
+  (or (eof-object? c) (opchar? c) (never-identifier-char? c)))
```

Comment:
I don't really understand this – is there a test case?

Reply:
Yes, the replcompletions test exercises this; the issue was that you need `Base.incomplete_tag(Meta.parse("\"\$foo", raise=false))` to return `:string`.

If you don't check `eof-object?` here, then parsing the incomplete string `"$foo` will raise a syntax error for because `$foo` is terminated by an eof, instead of raising an `incomplete` error.

Reply:
I suppose we could add an explicit `@test Base.incomplete_tag(Meta.parse("\"\$foo", raise=false)) == :string` too.

Reply:
Ah, makes sense. Can't hurt to have the explicit test here.